programs/zk-token-proof/src/lib.rs

```diff
@@ -24,39 +22,46 @@ fn process_verify_proof<T: Pod + ZkProofData>(
     let instruction_context = transaction_context.get_current_instruction_context()?;
     let instruction_data = instruction_context.get_instruction_data();
 
+    // will not panic since the first two bytes are checked earlier by the caller
     let verify_proof_data = VerifyProofData::<T>::try_from_bytes(&instruction_data[1..])?;
+    let proof_data = &verify_proof_data.proof_data;
 
-    verify_proof_data.proof_data.verify_proof().map_err(|err| {
+    proof_data.verify_proof().map_err(|err| {
         ic_msg!(invoke_context, "proof_verification failed: {:?}", err);
         InstructionError::InvalidInstructionData
     })?;
 
-    if let Some(context_state_authority) = verify_proof_data.create_context_state_with_authority {
+    // create context state if accounts are provided with the instruction
+    if instruction_context.get_number_of_instruction_accounts() > 0 {
+        let context_state_authority = {
```

Comment:
Yep, it turns out that scoping is actually not needed here since we make a copy of the account pubkey and the borrowed account is immediately dropped afterwards. Hence I removed it.